### ISSUE TITLE
Question about ConvexGIoULossFuction()

### ISSUE THREAD

**Issue by HsLOL** [open-mmlab/mmrotate#249]
想问一下，在ConvexGIoULossFuction()函数中，有关梯度的计算是什么作用呢？

```
        ctx.save_for_backward(pred)
        convex_gious, grad = convex_giou(pred, target)

        loss = 1 - convex_gious
        if weight is not None:
            loss = loss * weight
            grad = grad * weight.reshape(-1, 1)
        if reduction == 'sum':
            loss = loss.sum()
        elif reduction == 'mean':
            loss = loss.mean()

        unvaild_inds = torch.nonzero((grad > 1).sum(1), as_tuple=False)[:, 0]
        grad[unvaild_inds] = 1e-6

        # _reduce_grad
        reduce_grad = -grad / grad.size(0) * loss_weight
        ctx.convex_points_grad = reduce_grad
        return loss
```

**Comment by zytx121** [comment]
This is to enable the gradient of loss can be  back propagation, so as to optimize the model parameters

**Comment by HsLOL** [comment]
> 

I understood，Thanks for you reply ~
